src/workerd/api/node/node.h

```diff
@@ -31,15 +32,27 @@ class CompatibilityFlags : public jsg::Object {
   }
 };
 
+enum class NodeJsCompatAccess {
+  // Require `nodejs_compat` flag to be enabled for access to any Node module
+  REQUIRE_FLAG_FOR_ALL,
+  // Require `nodejs_compat` flag to be enabled only for user facing modules,
+  // always register internal modules (e.g. `node-internal:internal_inspect` for logging)
+  REQUIRE_FLAG_FOR_EXTERNAL_ONLY,
+};
```

Comment:
I wonder if this is actually necessary. We can just make the `REQUIRE_FLAG_FOR_EXTERNAL_ONLY` be the only supported behavior here and just make the Node.js modules always available for built-ins even without the flag.